### PR TITLE
[LOOP-4622] Scenario loading without access to health

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/LOOP-4622

### CODE REVIEW

File: Loop/Managers/DeviceDataManager.swift [hunk 1]
```diff
@@ -1241,6 +1241,14 @@ extension DeviceDataManager {
                 return
             }
 
+            guard !self.doseStore.sharingDenied else {
+                // only clear cache since access to health kit is denied
+                insulinDeliveryStore.purgeCachedInsulinDeliveryObjects() { error in
```
Comment (ps2) [inline Loop/Managers/DeviceDataManager.swift:1245]: This does more than just clear HK. It's main function is to clear the doses from DoseStores's CoreData based InsulinDeliveryStore, and clearing HK should be a side effect. I'm guessing if there's a problem resetting when HK sharing is denied, that the solution is going to be in LoopKit.

File: Loop/Managers/DeviceDataManager.swift [hunk 2]
```diff
@@ -1256,6 +1264,14 @@ extension DeviceDataManager {
             assertionFailure("\(#function) should be invoked only when a testing CGM manager is in use")
             return
         }
+        
+        guard !glucoseStore.sharingDenied else {
+            // only clear cache since access to health kit is denied
+            glucoseStore.purgeCachedGlucoseObjects() { error in
```
Comment (ps2) [inline Loop/Managers/DeviceDataManager.swift:1268]: Similar situation here, to the one described above.

**Review by ps2 — CHANGES_REQUESTED** [review]
I think this is probably leaving doses and samples around, and not really clearing everything.

File: Loop/Managers/DeviceDataManager.swift (hunk 2 above)
Comment (nhamming) [reply Loop/Managers/DeviceDataManager.swift:1268]: I think I’m missing something. For `GlucoseStore` `purgeAllGlucoseSamples` it seems there are only 2 actions:
- `purgeCachedGlucoseObjects`
- `healthStore.deleteObjects`
Is there an additional action related to `CoreData` that I’m overlooking?

Note: there is also a clean-up step `handleUpdatedGlucoseData` but that step is also apart of `purgeCachedGlucoseObjects`

Comment (ps2) [reply Loop/Managers/DeviceDataManager.swift:1268]: Oh, I think I misread the code. This is purging core data (via purgeCachedGlucoseObjects) inside the guard when not sharing, and when sharing, it skips purgeCachedGlucoseObjects, and then deletes HK data directly.

In the case where sharing is allowed, are you relying on the HK deletes to delete all the data that `purgeCachedGlucoseObjects` would be deleting from CoreData?
 
As an aside, there are a lot of negatives in these two conditions! Hard to keep track.

sharingDenied == not sharing
!sharingDenied == not not sharing
guard !sharingDenied else {do it} == if not not not sharing {do it}

**Review by ps2 — APPROVED** [review]
LGTM

**Comment by nhamming** [comment]
https://tidepool.atlassian.net/browse/COASTAL-1239 will handle the correction to only delete testing device data.

**Pull request merged**